packages/store-sdk/src/storage/useStorage.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * Safe IDB storage interface. These try..catch are usefull because
+ * some browsers may block accesss to these APIs due to security policies
+ *
+ * Also, the stored value is lazy-loaded to avoid hydration mimatch
+ * between server/browser. When state is 'hydrated', the value in the heap
+ * is the same as the value in IDB
+ */
+import { useState, useEffect, useMemo } from 'react'
+import { get, set } from 'idb-keyval'
+
+const getItem = async <T>(key: string) => {
+  try {
+    const value = await get<T>(key)
+
+    return value ?? null
+  } catch (err) {
+    return null
+  }
+}
+
+const setItem = async <T>(key: string, value: T | null) => {
+  try {
+    await set(key, value)
```

Comment:
Do we really need to `await` here? Maybe we can just ignore that Promise since if it fails we aren't going to do anything anyways.

Reply:
but then we will get an unhandled promise rejection being logged into our sentry, right?

Reply:
You're right!